applications/vanilla/controllers/class.vanillasettingscontroller.php

```diff
@@ -40,11 +40,25 @@ public function posting() {
         // Check permission
         $this->permission('Garden.Settings.Manage');
 
+        $formats = ['Text']; // Check for additional formats
+        $this->EventArguments['formats'] = &$formats;
+
+        $this->fireEvent('GetFormats');
+        // This was moved from the editor plugin. In order to maintain compatibility with existing event hooks
+        // We will need to fire this as EditorPlugin.
+        $this->EventArguments['formats'] = &$formats;
+        $this->fireAs('EditorPlugin');
+        $this->fireEvent('GetFormats');
```

Comment:
The event manager has a `fireDeprecated` method that might be useful here. It'll trigger deprecation notices if any handlers are detected hooking into this event.